### PR TITLE
Cover/cmds/core/chroot: Add tests.

### PR DESCRIPTION
**Pull request by llogen**
Refactored chroot.go and chroot_test.go and added some tests.

Signed-off-by: llogen <christophlange95@googlemail.com>

### CODE REVIEW

File: cmds/core/chroot/chroot.go [hunk 1]
```diff
@@ -149,39 +149,36 @@ func isRoot(dir string) (bool, error) {
...
-		err       error
 	)
 
-	flag.Parse()
-
-	if flag.NFlag() == 0 && flag.NArg() == 0 {
+	if flag.NFlag() == 0 && len(args) == 0 {
 		flag.PrintDefaults()
-		os.Exit(1)
+		return nil
```
[the remark this answers is not in the document]
Comment (llogen) [reply cmds/core/chroot/chroot.go:159]: Due to the fact that we are using flag.PrintDefaults(), it is not possible. I set the output of the Print to a buffer to retrieve the printing for testing. Does this satisfy you or was your intention something else?

File: cmds/core/chroot/chroot.go [hunk 2]
```diff
@@ -196,6 +192,14 @@ func main() {
...
 
 	if err = cmd.Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	if err := chroot(flag.NArg(), flag.Args()...); err != nil {
```
Comment (rminnich) [inline cmds/core/chroot/chroot.go:202]: I don't see the reason for passing flag.Narg() given that we can get Nargs from len(flag.Args())

File: cmds/core/chroot/chroot.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/core/chroot/chroot.go:159]: see flag.CommandLine.SetOutput

[the remark this answers is not in the document]
Comment (llogen) [reply cmds/core/chroot/chroot.go:159]: I don't think that i understand properly. The question was if the usage of flag.CommandLine.SetOutput in the testfile will satisfy you? Like this we can retrieve the flag.PrintDefaults() for testing purposes.

File: cmds/core/chroot/chroot.go (hunk 2 above)
Comment (llogen) [reply cmds/core/chroot/chroot.go]: I was confused about this. Fixed it now. Thank you for catching it.

File: cmds/core/chroot/chroot.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/core/chroot/chroot.go:151]: That would be helpful.

[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/core/chroot/chroot.go:151]: but overall this is kind of messy. a function that returns an err, that will be printed by the caller, is also printing things, and there is no guarantee that the way that one print is done will match with how the caller prints the rest of it. It seems very awkward. If would be nicer in my view of we had a way to get the PrintDefaults output  into a string. '
This is arguably a deficiency in the flag package which is really early code (2008? earlier?) but we're not able to change its deficiences.

[the remark this answers is not in the document]
Comment (llogen) [reply cmds/core/chroot/chroot.go:154]: Now i created a defaults string that will be printed.

**Review by rminnich — DISMISSED** [review]

**Review by ChriMarMe — APPROVED** [review]

**Pull request merged**